cmd/kubetype-gen/generators/types.go

```diff
@@ -61,6 +62,12 @@ func (g *typesGenerator) GenerateType(c *generator.Context, t *types.Type, w io.
 		"IstioStatus": c.Universe.Type(types.Name{Name: "IstioStatus", Package: "istio.io/api/meta/v1alpha1"}),
 	}
 	for _, kubeType := range kubeTypes {
+		localM := m
+		// ServiceEntry has a unique status type which includes addresses for auto allocated IPs, substitute IstioServiceEntryStatus
+		// for IstioStatus when type is ServiceEntry
+		if slices.Contains(kubeType.RawType().CommentLines, useIstioServiceEntryStatus) {
+			localM["IstioStatus"] = c.Universe.Type(types.Name{Name: "IstioServiceEntryStatus", Package: "istio.io/api/meta/v1alpha1"})
```

Comment:
nit: extract the custom type instead of hard coding .

Reply:
added, also https://github.com/istio/api/pull/3257 adds the required comment so we can remove the hardcode